docker/development.Dockerfile

```diff
@@ -16,26 +16,27 @@ RUN apt-get install -y python-software-properties \
 	libidn11 \
 	git
 
-ADD https://cmake.org/files/v3.3/cmake-3.3.2-Linux-x86_64.tar.gz /opt/
-RUN tar -C /opt -xvzf /opt/cmake-3.3.2-Linux-x86_64.tar.gz
-RUN cp -r /opt/cmake-3.3.2-Linux-x86_64/* /usr
-
-ADD https://launchpad.net/ubuntu/+archive/primary/+files/snappy_1.1.3.orig.tar.gz /opt/
-RUN tar -C /opt -xf /opt/snappy_1.1.3.orig.tar.gz
-RUN mkdir /opt/snappy-1.1.3/build && cd /opt/snappy-1.1.3/build && ../configure && make && make install
-
-ADD https://openssl.org/source/openssl-1.0.2k.tar.gz /opt/
+RUN curl -L -o /opt/openssl-1.0.2k.tar.gz https://openssl.org/source/openssl-1.0.2k.tar.gz
```

Comment:
Any reason this can't be an `ADD` (vs `RUN`)? When using `ADD` it uses the host machine's internet connection and copies the files into the container. That way we don't need to have `curl` installed inside the docker container.

Reply:
The problem with `ADD` is that it does not cache the URL fetches: https://github.com/moby/moby/issues/880

So all download steps run again and again when you want to do a build and test. It's recommended that we use `curl` if we want to cache downloads.